cmd/cloud_sql_proxy/proxy.go

```diff
@@ -233,11 +233,19 @@ func parseInstanceConfig(dir, instance string, cl *http.Client) (instanceConfig,
 			// No "host" part of the address. Be safe and assume that they want a
 			// loopback address.
 			ret.Network = spl[0]
-			addr, ok := loopbackForNet[spl[0]]
-			if !ok {
-				return ret, fmt.Errorf("invalid %q: unrecognized network %v", instance, spl[0])
+			if spl[0] == "unix" {
+				if strings.HasPrefix(spl[1], "/") {
+					ret.Address = spl[1]
+				} else {
+					ret.Address = filepath.Join(dir, spl[1])
```

Comment:
do we want to append `".s.PGSQL.5432"` for postgres instances?

Reply:
I have no experience with google postgres instances so far. Is this `.s.PGSQL.5432` string appended to the socket name when postgres is used? This shouldn't happen as this pull request is about setting a completely custom socket name when one cannot (or doesn't want to) change the behaviour of some third party software.

Reply:
PostgreSQL treats the unix socket location as a directory, and specifies the socket name should be `.s.PGSQL.5432` - see "unix_socket_directories" [section](https://www.postgresql.org/docs/11/runtime-config-connection.html). The proxy currently automatically appends this suffix [here](https://github.com/GoogleCloudPlatform/cloudsql-proxy/blob/6ac0b49e7197a3f91a3d438204a45e0032af2de9/cmd/cloud_sql_proxy/proxy.go#L276).

Most client side connectors treat it as a directory and automatically append the `.s.PGSQL.5432` automatically, which means if the user specifies `/my/socket/file`, it is actually using `/my/socket/file/.s.PGSQL.5432`. Since we already automatically append in one scenario, it seems best that we should continue to do so for this one to remain consistent. 

It might make more sense to move the append into the file creation to consolidate, which I think is [here](https://github.com/GoogleCloudPlatform/cloudsql-proxy/blob/dc8853bb2cd44140b12bffdaa5603aa99647fcd5/cmd/cloud_sql_proxy/proxy.go#L136).